docs/design/Path concatenation.md

```diff
@@ -0,0 +1,94 @@
+## Paths in DF2
+
+It is expected that the use of paths in DF2 (similar in nature to Unix file paths, but not exactly the same) will be extensive, as will be concatention operations on them: `GetFullPath` and `GetRelativePath`. This document will provide a specification for how these operations are to behave as well as contain some considerations that will drive the design of concatenation functions.
+
+## Definitions
+
+_Char_ is a Unicode code point. There are special _chars_ which have specific meaning in DF2 paths:
+- _Separator_ is a _char_ corresponding to the code point `U+002F` - `Slash`.
+- _Dot_ is a _char_ corresponding to the code point `U+002E` - `Full stop`.
+
+Two _chars_ are equal if they refer to the same code point.
+
+[Rationale: to support the performance of concatenation functions, it is neccessary to impose the restriction that equality comparisons be strictly ordinal.]
+
+_Segments_ are sequences of _chars_ that have no _separators_ in them. There are special _segments_ which have specific meaning in DF2 paths:
+- _Skip segment_ is a _segment_ that consists of one _dot_ or has no _chars_ at all.
```

Comment:
for what reason are dots allowed? seems pointless to me.

Especially due to the fact that this complicates checking paths for equality.

Reply:
I am actually not sure of that either, @Perksey is the person to ask (my understanding was it was initially for user convenience, but that seems more like a UI-level thing).

Reply:
Mainly for consistency across other path specs. We can remove if there's majority agreement, just means that a user trying ./ may get unexpected result but as you said that's UI-level and can easily be worked around.

Reply:
I am personally fine with leaving this in given in-process all paths will be normalized anyway.

Reply:
in that case I'd suggest adding normalized paths have to bestripped of the redundant dots.

Reply:
I believe that's already the case:
> For each skip segment in the path, delete it and the separator preceding it.

Reply:
Right, sorry. Can't resolve comments here unfortunately, but I'd mark this as resolved.